docker.go

```diff
@@ -1364,13 +1364,17 @@ func (p *DockerProvider) ReuseOrCreateContainer(ctx context.Context, req Contain
 		lifecycleHooks:    []ContainerLifecycleHooks{combineContainerHooks(defaultHooks, req.LifecycleHooks)},
 	}
 
-	if p.config.RyukDisabled {
-		// ryuk disabled, ensure the container we are about to reuse is running
-		if c.State != "running" {
-			// found container to reuse, but it is not in state running
-			if err := dc.Start(ctx); err != nil {
-				return dc, fmt.Errorf("start container %s in state %s: %w", req.Name, c.State, err)
-			}
+	// If a container was stopped programmatically, and Ryuk is enabled, we want to
+	// ensure the container is running again, but only if it is not paused, as it's
+	// not possible to start a paused container. The Docker Engine returns the
+	// "cannot start a paused container, try unpause instead" error.
+	switch c.State {
+	case "running", "paused":
```

Comment:
question: should we unpause it or error?

Reply:
I added the //TODO message

Reply:
Would it be better to just error for paused, indicating it's not supported, that way it will fail fast?

Reply:
Yeah, I like the idea, thanks!